docs/source/guides/speed.rst

```diff
@@ -0,0 +1,279 @@
+        module avail
+
+        # load correct CUDA before install
+        module load cuda-10.0
+        # ------------------------
+
+        # make sure you've loaded a cuda version > 4.0 and < 7.0
+        module load gcc-6.1.0
+
+        $ pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" https://github.com/NVIDIA/apex
+
+    2. Set the `precision` trainer flag to 16. You can customize the `Apex optimization level <https://nvidia.github.io/apex/amp.html#opt-levels>`_ by setting the `amp_level` flag.
+
+    .. testcode::
+        :skipif: not _APEX_AVAILABLE and not _NATIVE_AMP_AVAILABLE or not torch.cuda.is_available()
+
+        # turn on 16-bit
+        trainer = Trainer(amp_backend="apex", amp_level='O2', precision=16)
+
+    If you need to configure the apex init for your particular use case, or want to ucustumize the
+    16-bit training behviour, override :meth:`pytorch_lightning.core.LightningModule.configure_apex`.
+
+16-bit precision on TPUs
+========================
+To use 16-bit precision on TPUs simply set the number of tpu cores, and set `precision` trainer flag to 16.
+
+.. testcode::
+    :skipif: not _TPU_AVAILABLE
+
+    # DEFAULT
+    trainer = Trainer(tpu_cores=8, precision=32)
+
+    # turn on 16-bit
+    trainer = Trainer(tpu_cores=8, precision=16)
+
+----------------
+
+
+***********************
+Control Training epochs
+***********************
+
+It can be useful to force training for a minimum number of epochs or limit to a max number of epochs. Use the `min_epochs` and `max_epochs` Trainer flags to set the number of epochs to run.
+
+.. seealso:: :class:`~pytorch_lightning.trainer.trainer.Trainer`
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(min_epochs=1, max_epochs=1000)
+
+----------------
+
+****************************
+Control validation frequency
+****************************
+
+Check validation every n epochs
+===============================
+If you have a small dataset, you might want to check validation every n epochs. Use the `check_val_every_n_epoch` Trainer flag.
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(check_val_every_n_epoch=1)
+
+
+Set validation check frequency within 1 training epoch
+======================================================
+For large datasets, it's often desirable to check validation multiple times within a training loop.
+Pass in a float to check that often within 1 training epoch. Pass in an int `k` to check every `k` training batches.
+Must use an `int` if using an `IterableDataset`.
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(val_check_interval=0.95)
+
+    # check every .25 of an epoch
+    trainer = Trainer(val_check_interval=0.25)
+
+    # check every 100 train batches (ie: for `IterableDatasets` or fixed frequency)
+    trainer = Trainer(val_check_interval=100)
+
+----------------
+
+******************
+Limit dataset size
+******************
+
+Use data subset for training, validation, and test
```

Comment:
Explain how this leads to faster convergence

Reply:
I'm assuming large datasets but idk, feel free to add